### PR TITLE
[Dashboard] Token balance conversion to CLNY token

### PR DESCRIPTION
**Pull request by mmioana**
## Description

Currently we try to convert a token by it's id to the selected currency which can be `CLNY`. However, `CoinGecko` doesn't provide this information in the `${COINGECKO_API_URL}/api/v3/coins/${tokenId}/history` endpoint response.
So the easy fix is to get all balance data in `USD` and then do the conversion from `USD` to `CLNY`

This PR addresses this issue, plus a bigger refactoring which introduces 3 contexts in order to avoid querying the same data multiple times.

## Testing

TODO: Please test the data displayed in the `Total` cards and the `Total in and out` card are updating properly based on the selected currency.

Screenshot 2024-10-01 at 17 04 41
Screenshot 2024-10-01 at 17 06 06

## Diffs

**New stuff** ✨

* `BalanceCurrencyContext`, `TotalInOutBalanceContext` and `TotalInOutBalanceChartContext` contexts
* `convertFromTokenToClny`

**Deletions** ⚰️

* `usePreviousLast30DaysData`
* `useLast30DaysData`
* `useData`
* `useChartYSteps`
* Unused `useTotalFunds` and `calculateTotalFunds`

Resolves [#3226](https://github.com/JoinColony/colonyCDapp/issues/3226)

### CODE REVIEW

**Review by bassgeta — APPROVED** [review]
fur such large change blocks, this is very readable and it makes a lot of sense when you go through it, so great job overall!
Loving the fact that we use a context to dedupe calls for this hefty lambda, optimization :chart_with_upwards_trend: 

Fingers crossed this is gonna be one of the last things you need to do on this giga chonker :sweat_smile: 

All teams:
image
Switching to ETH shows up loaders:
image
Gets updated (ofc with dummy values locally)
image
CLNY also works:
image

:ship:

File: src/context/BalanceCurrencyContext/BalanceCurrencyContextProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+const BalanceCurrencyContextProvider = ({
+  children,
+}: {
+  children: ReactNode;
+}) => {
+  const { currency } = useCurrencyContext();
+  const queryCurrency =
+    currency === SupportedCurrencies.Clny ? SupportedCurrencies.Usd : currency;
+
+  const defaultClnyInUSD = isDev ? 1 : 0;
```
Comment (davecreaser) [inline src/context/BalanceCurrencyContext/BalanceCurrencyContextProvider.tsx:26]: Very handy, thank you 😆

File: src/components/v5/frame/ColonyHome/partials/FundsCards/hooks.ts [hunk 2]
```diff
@@ -1,62 +1,115 @@
...
+  }, [selectedDomain?.id, colonyAddress]);
+
+  const { data, loading } = useGetCachedDomainBalanceQuery(
+    memoizedQueryVariables.queryOptions,
+  );
+
+  useEffect(() => {
+    return () => {
+      if (loading) {
+        memoizedQueryVariables.abortController.abort();
```
Comment (davecreaser) [inline src/components/v5/frame/ColonyHome/partials/FundsCards/hooks.ts:109]: 👏 Really nice fix for this domain problem!

**Review by davecreaser — APPROVED** [review]
Wow! A lot of work here 😅 Really good stuff, I think the contexts are a neat idea, and it certainly made that hook file smaller!

I tried my best to understand everything that was written since this is a large, complex PR, but I think you've done a really good job with it from what I can tell. 🎉 

Also tested it, and it worked as expected:
Screenshot 2024-10-03 at 16 41 32
Screenshot 2024-10-03 at 16 41 46

**Pull request merged**